core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -301,6 +331,12 @@ public boolean maybeAcquireFetchLock() {
      * @return A future which is completed when the records are acquired.
      */
     public CompletableFuture<List<AcquiredRecords>> acquire(String memberId, FetchPartitionData fetchPartitionData) {
+        log.trace("Received acquire request for share partition: {}-{}", memberId, fetchPartitionData);
+        if (!isReady.get()) {
+            log.error("Share partition not completed initilization");
+            return FutureUtils.failedFuture(new IllegalStateException("Share partition is not ready to perform operations"));
+        }
+
```

Comment:
@AndrewJSchofield This might fail the first request. So do you think it might be reasonable 1) to wait for the future to complete in initilization or 2) create another method for initilization which return CompletableFuture and can be chained in SharePartitionManager?

Reply:
Done.